src/main/java/com/epam/training/gen/ai/data/ChatWithHistoryResponse.java

```diff
@@ -0,0 +1,9 @@
+package com.epam.training.gen.ai.data;
+
+import java.util.List;
+import java.util.UUID;
+
+public record ChatWithHistoryResponse(
+        UUID chatId,
+        List<ChatRoleMessage> messages
```

Comment:
I just wonder if it would be enough to just reply with a String object instead of the ChatRoleMessage instance as it may contain more information. I guess it depends on needs or requirements and how you visualize your app :D